src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,153 @@
 import type { FC } from "react";
 import { useNotify, Spinner } from "@canonical/react-components";
-import { useQuery } from "@tanstack/react-query";
-import { fetchResources } from "api/server";
-import { queryKeys } from "util/queryKeys";
 import { humanFileSize } from "util/helpers";
 import { limitToBytes } from "util/limits";
 import type { LxdProject } from "types/project";
-import { useServerEntitlements } from "util/entitlements/server";
+import type { LxdResources } from "types/resources";
+import { Icon } from "@canonical/react-components";
+import type { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import type { CreateInstanceFormValues } from "pages/instances/CreateInstance";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { useClusterMembers } from "context/useClusterMembers";
+import { CLUSTER_GROUP_PREFIX } from "util/instances";
+import ResourceLink from "components/ResourceLink";
+import { useIsClustered } from "context/useIsClustered";
+import {
+  useUnClusterMemberResources,
+  useClusterMemberResources,
+} from "context/useResources";
 
 interface Props {
+  formik?: InstanceAndProfileFormikProps;
   project?: LxdProject;
 }
 
-const MemoryLimitAvailable: FC<Props> = ({ project }) => {
+const MemoryLimitAvailable: FC<Props> = ({ formik, project }) => {
   const notify = useNotify();
-  const { canViewResources } = useServerEntitlements();
+  const isClustered = useIsClustered();
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const values = formik?.values;
+  const projectResourceMemory = limitToBytes(project?.config["limits.memory"]);
+  const currentServer = (() => {
+    const createTarget =
+      (values as CreateInstanceFormValues)?.target ?? undefined;
+    if (createTarget && !createTarget.startsWith(CLUSTER_GROUP_PREFIX)) {
+      return createTarget;
+    }
+    const location = (values as EditInstanceFormValues)?.location ?? undefined;
+    return location === "none" || !location ? undefined : location;
+  })();
+
+  const {
+    data: clusteredResources = [],
+    error: clusteredError,
+    isLoading: isClusteredLoading,
+  } = useClusterMemberResources(
+    clusterMembers.filter(
+      (member) => !currentServer || member.server_name === currentServer,
+    ),
+  );
 
   const {
-    data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
-  });
-
-  if (isLoading) {
+    data: unClusteredResource,
+    error: unClusteredError,
+    isLoading: isUnClusteredLoading,
+  } = useUnClusterMemberResources();
```

Comment:
This is pretty nice already. I wonder if can combine the data loding to pass in the currentServer into a `useResources` function and it internally loads the cluster members if server is clustered and picks the right query between clustered and unclustered cases. That should simplify the rendering logic in here as we can always refer to the same resources array for both cases. So we remove the conditions between clusteredResources and unClusteredResource